src/runtime_src/core/tools/xbmgmt2/OO_Config.cpp

```diff
@@ -19,32 +19,257 @@
+    ("show", boost::program_options::bool_switch(&m_show), "<add description>")
+    ("enable_retention", boost::program_options::bool_switch(&m_enable_retention), "<add description>")
+    ("disable_retention", boost::program_options::bool_switch(&m_disable_retention), "<add description>")
+    ("ddr", boost::program_options::bool_switch(&m_ddr), "<add description>")
+    ("hbm", boost::program_options::bool_switch(&m_hbm), "<add description>")
     ("help,h", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
   ;
 }
 
 void
 OO_Config::execute(const SubCmdOptions& _options) const
 {
-  _options.size();
-  printHelp();
+
+  XBU::verbose("SubCommand option: config");
+
+  XBU::verbose("Option(s):");
+  for (auto & aString : _options)
+    XBU::verbose(std::string(" ") + aString);
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).run(), vm);
+    po::notify(vm); // Can throw
+  } catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+    printHelp();
+    throw; // Re-throw exception
+  }
+
+  //exit if neither daemon or device is specified
+  if(m_help || (m_device.empty() && !m_daemon)) { 
+    printHelp();
+    return;
+  }
+  
+  // parse device indices
+  std::vector<uint16_t> device_indices;  
+  XBU::parse_device_indices(device_indices, m_device);
+  
+  //Option:show
+  if(m_show) {
+    XBU::verbose("Sub command: --show");
+    //show daemon config
+    if(m_daemon)
+      show_daemon_conf();
+
+    //show device config
+    if (!device_indices.empty()) {
+      for(auto idx : device_indices) {
+        auto dev = xrt_core::get_mgmtpf_device(idx);
+        show_device_conf(dev);
+      }
+    }
+  return;
+  }
+
+  //Option:daemon
+  if(m_daemon) {
+    XBU::verbose("Sub command: --daemon");
+    if(m_host.empty())
+      throw xrt_core::error("Please specify ip or hostname for peer");
+    update_daemon_config(m_host);
+    return;
+  }
+
+  //Option:device
+  if(!m_device.empty()) {
+    XBU::verbose("Sub command: --device");
+    //update security
+    if (!m_security.empty()) {
+      for(auto idx : device_indices) {
+        auto dev = xrt_core::get_mgmtpf_device(idx);
+        update_device_conf(dev, m_security, configType::security);
+      }
+    }
+
+    //clock scaling
+    if (!m_clk_scale.empty()) {
+      for(auto idx : device_indices) {
+        auto dev = xrt_core::get_mgmtpf_device(idx);
+        update_device_conf(dev, m_clk_scale, configType::clk_scaling);
+      }
+    }
+    
+    //update threshold power override
+    if (!m_power_override.empty()) {
+      for(auto idx : device_indices) {
+        auto dev = xrt_core::get_mgmtpf_device(idx);
+        update_device_conf(dev, m_power_override, configType::cs_threshold_power_override);
```

Comment:
Take it or leave it, but I think the mixed case doesn't look nice with the style otherwise used.  I'd by far prefer enum class config_type over the camel case.

Reply:
Sorry! missed that